[ADD] accounting: adding polish localization (#18147)
odoo/documentation: content/applications/finance/fiscal_localizations/poland.rst

   <general/install>`.

.. _localizations/poland/ksef:

E-invoicing with KSeF
=====================

`KSeF <https://ksef.podatki.gov.pl/>`_ is Poland's mandatory, centralized electronic invoicing
system used to issue, transmit, and store electronic invoices according to national regulations.

.. _localizations/poland/ksef-configuration:

.. seealso::
   `KSeF platform video tutorials of the Polish Ministry of Finance <https://www.youtube.com/watch?v=V6E1fGUFPw4&list=PLGj5OAFBHLKX0cEVGjcGplnbX6LpDFF38&index=1>`_

Configuration
-------------

To use KSeF e-Invoicing with Odoo, first `create a KSeF account <https://ap.ksef.mf.gov.pl/web/>`_
or log in using your company's NIP if you already have credentials. Next, request a certificate and
ensure you save the password created during this process, as it will be required later.

Once the certificate request is complete, download the `.pem` **certificate file** and save the
**private key** issued by KSeF. You will need both to complete the Odoo integration.

.. _localizations/poland/linking-ksef:

Link Odoo to KSeF
~~~~~~~~~~~~~~~~~

.. seealso::

>    KSeF test and pre-production environments are available `here <https://ksef.podatki.gov.pl/aplikacja-podatnika-ksef-20/>`_

Reviewer: ```suggestion
   `KSeF test and pre-production environments <https://ksef.podatki.gov.pl/aplikacja-podatnika-ksef-20/>`_
```
We try to have [descriptive hyperlinks](https://www.odoo.com/documentation/19.0/contributing/documentation/rst_guidelines.html#hyperlinks), avoiding the [click here](https://www.odoo.com/documentation/19.0/contributing/documentation/rst_guidelines.html#hyperlinks) kind.

   * - :guilabel:`Poland - Accounting - Bank Account Verification`
     - `l10n_pl_bank_verification`
     - Verification of the VAT/Bank account number combination for PL-to-PL payments exceeding
       15,000 PLN
   * - :guilabel:`Poland - JPK FA Reports`
     - `l10n_pl_reports_jpk_fa`
     - Module to generate JPK_FA XML reports for Poland
   * - :guilabel:`Poland - JPK_VAT PoS Enterprise`
     - `l10n_pl_reports_pos_jpk`
     - Integration module for PoS information in the Polish JPK report
   * - :guilabel:`Poland - SAFT Income Tax Report (JPK KR PD)`
     - `l10n_pl_reports_account_saft`
     - Module to generate the JPK_KR_PD XML report for Poland

.. note::
   In some cases, such as when upgrading to a version with additional modules, it is possible that
   modules may not be installed automatically. Any missing modules can be manually :ref:`installed
   <general/install>`.

.. _localizations/poland/ksef:

E-invoicing with KSeF
=====================

`KSeF <https://ksef.podatki.gov.pl/>`_ is Poland's mandatory, centralized electronic invoicing
system used to issue, transmit, and store electronic invoices according to national regulations.

> 

Reviewer: ```suggestion
```

.. _localizations/poland/ksef-configuration:

Configuration
-------------

.. tip::
   The Polish Ministry of Finance provides a `video documentation playlist <https://www.youtube.com/watch?v=V6E1fGUFPw4&list=PLGj5OAFBHLKX0cEVGjcGplnbX6LpDFF38&index=1>`_

>    on the KSeF platform.

Reviewer: ```suggestion
```
I would transform this into a `seealso` and move it above the configuration heading, something like:

```rst
.. seealso::
   `KSeF platform video tutorials of the Polish Ministry of Finance <https://www.youtube.com/watch?v=V6E1fGUFPw4&list=PLGj5OAFBHLKX0cEVGjcGplnbX6LpDFF38&index=1>`_
```

To use KSeF e-Invoicing with Odoo, first `create a KSeF account <https://ap.ksef.mf.gov.pl/web/>`_
or log in using your company's NIP if you already have credentials. Next, request a certificate and
ensure you save the password created during this process, as it will be required later.

> Once the certificate request is complete, download the **certificate file (`.pem`)** and save the

Reviewer: ```suggestion
Once the certificate request is complete, download the `.pem` **certificate file** and save the
```
The literal markup doesn't work within the bold markup, so you could move it before/after.
<img width="187" height="37" alt="Image" src="https://github.com/user-attachments/assets/9719ac12-54d7-449f-a514-c4d6e8207487" />

**private key** issued by KSeF. You will need both to complete the Odoo integration.

.. _localizations/poland/linking-ksef:

Link Odoo to KSeF
~~~~~~~~~~~~~~~~~

.. tip::

>    KSeF test and pre-production environments are available `here <https://ksef.podatki.gov.pl/aplikacja-podatnika-ksef-20/>`_.

Reviewer: ```suggestion
.. seealso::
   `KSeF test and pre-production environments <https://ksef.podatki.gov.pl/aplikacja-podatnika-ksef-20/>`_.
```
Unless you give more info about the test/pre-prod environment and how they can be used, I wouldn't consider this a tip, but a seealso.

#. Open the **Accounting** app and navigate to :menuselection:`Configuration --> Settings`.
#. Scroll down to the :guilabel:`Polish Localization` section and enable :guilabel:`Allow KSeF
   integration`.
#. Enter a name for your KSeF certificate in the newly enabled field, then select :guilabel:`Create
   and edit...` from the drop-down menu.
#. In the pop-up window, click :guilabel:`Upload your file` and select your **certificate**

>    (`.rpm`). In the :guilabel:`Certificate Password` field, enter the password created during the

Reviewer: ```suggestion
   (`.pem`). In the :guilabel:`Certificate Password` field, enter the password created during the
```

   certificate application on the KSeF platform.
#. Once the certificate is uploaded, the :guilabel:`Private Key` field appears. Click the empty
   field, select :guilabel:`Create` from the drop-down menu, and enter your **private key**. The
   **public key** is embedded within the `.pem` file and is automatically filled in.

>    Finally, click :guilabel:`Save`.

Reviewer: ```suggestion
#. Finally, click :guilabel:`Save`.
```

send invoices to KSeF directly through Odoo:

#. Open the **Accounting** app, go to :menuselection:`Customers --> Invoices`, and select a posted
   invoice.
#. Click :guilabel:`Send`.
#. In the pop-up window, check :guilabel:`Send via KSeF (e-Faktura)` and click :guilabel:`Send`.

When an invoice is sent, Odoo automatically:

- Generates the invoice in the required FA(3) XML format.
- Submits the file to KSeF for validation.
- Saves a copy of the XML file in the invoice's chatter.

You can also manually export the FA(3) XML and check the invoice's processing status:

#. Open a posted invoice.
#. Click :guilabel:`Export FA(3) XML` to download the invoice as an XML file to your browser's
   download folder.
#. To check its processing status, click :guilabel:`Check invoice status` next to the export option.
#. The live status is displayed on the invoice under the :guilabel:`eInvoice status` field.

Once the Polish government officially registers the invoice on the KSeF platform, it assigns an
**Official Receipt Certificate (UPO)**. To view it, open the invoice and click
:guilabel:`Check invoice UPO`.

.. _localizations/poland/ksef-import:

Automatic bill imports

> ~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Automatic bill import
~~~~~~~~~~~~~~~~~~~~~
```
Singular sounds more correct, but I could be wrong

By default, the Polish localization automatically fetches vendor bills from KSeF at regular

> intervals. To adjust this frequency or toggle the feature:

Reviewer: ```suggestion
By default, the Polish localization automatically fetches vendor bills from KSeF every three hours. To adjust this frequency or toggle the feature:
```

#. Activate :doc:`developer mode <../../general/developer_mode>`.
#. Open the **Settings** app.
#. Click the :guilabel:`Technical` menu at the top and select :guilabel:`Automation: Scheduled

>    Actions`.

Reviewer: ```suggestion
#. With :doc:`developer mode <../../general/developer_mode>` activated, open the **Settings** app and 
     go to :menuselection:`Technical --> Scheduled Actions`
```
It could be a single "action" IMO

> #. Search for and open :guilabel:`5 Polish eInvoice: Download vendor bills from KSeF`.

Reviewer: ```suggestion
#. Search for and open :guilabel:`Polish eInvoice: Download vendor bills from KSeF`.
```

> #. Adjust the configuration according to your needs.

Reviewer: ```suggestion
#. Use the :guilabel:`Execute Every` field to change the frequency.
```
